pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -68,10 +70,29 @@ def __call__(self, *targs, **tkwargs):
         if callable(obj):
             return self._qfunc_transform(obj, targs, tkwargs)
 
-        raise TransformError(
-            "The object on which the transform is applied is not valid. It can only be a tape, a QNode or a qfunc."
+        # Input is not a QNode nor a quantum tape nor a device.
+        # Assume Python decorator syntax:
+        #
+        # result = some_transform(*transform_args)(qnode)(*qnode_args)
+
+        warnings.warn(
+            "The decorator syntax transform_fn(*transform_args)(qnode) has been "
+            "deprecated and will be removed in a future version. Please use either "
+            "transform_fn(qnode, *transform_args) or "
+            "functools.partial(transform_fn, *transform_args)(qnode) instead.",
+            UserWarning,
         )
 
+        if obj is not None:
+            targs = (obj, *targs)
+
+        def wrapper(obj):
+            return self(obj, *targs, **tkwargs)
+
+        wrapper.__doc__ = f"Partial of transform {self} with bound arguments and keyword arguments."
```

Comment:
does `self` have a good repr?

Reply:
Can we make it have a good repr?

Reply:
We could, but this involves input from @trbromley and @rmoyard and I don't want to block this PR.